// @remotion/serverless/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateComposition = exports.streamWriter = exports.ResponseStream = exports.stopLeakDetection = exports.setCurrentRequestId = exports.invokeWebhook = exports.innerHandler = exports.infoHandler = exports.stillHandler = exports.startHandler = exports.rendererHandler = exports.progressHandler = exports.launchHandler = exports.compositionsHandler = exports.getBrowserInstanceImplementation = exports.forgetBrowserEventLoopImplementation = void 0;
__exportStar(require("@remotion/serverless-client"), exports);
const get_browser_instance_1 = require("./get-browser-instance");
Object.defineProperty(exports, "forgetBrowserEventLoopImplementation", { enumerable: true, get: function () { return get_browser_instance_1.forgetBrowserEventLoopImplementation; } });
Object.defineProperty(exports, "getBrowserInstanceImplementation", { enumerable: true, get: function () { return get_browser_instance_1.getBrowserInstanceImplementation; } });
const compositions_1 = require("./handlers/compositions");
Object.defineProperty(exports, "compositionsHandler", { enumerable: true, get: function () { return compositions_1.compositionsHandler; } });
const launch_1 = require("./handlers/launch");
Object.defineProperty(exports, "launchHandler", { enumerable: true, get: function () { return launch_1.launchHandler; } });
const progress_1 = require("./handlers/progress");
Object.defineProperty(exports, "progressHandler", { enumerable: true, get: function () { return progress_1.progressHandler; } });
const renderer_1 = require("./handlers/renderer");
Object.defineProperty(exports, "rendererHandler", { enumerable: true, get: function () { return renderer_1.rendererHandler; } });
const start_1 = require("./handlers/start");
Object.defineProperty(exports, "startHandler", { enumerable: true, get: function () { return start_1.startHandler; } });
const still_1 = require("./handlers/still");
Object.defineProperty(exports, "stillHandler", { enumerable: true, get: function () { return still_1.stillHandler; } });
const info_1 = require("./info");
Object.defineProperty(exports, "infoHandler", { enumerable: true, get: function () { return info_1.infoHandler; } });
const inner_routine_1 = require("./inner-routine");
Object.defineProperty(exports, "innerHandler", { enumerable: true, get: function () { return inner_routine_1.innerHandler; } });
const invoke_webhook_1 = require("./invoke-webhook");
Object.defineProperty(exports, "invokeWebhook", { enumerable: true, get: function () { return invoke_webhook_1.invokeWebhook; } });
const leak_detection_1 = require("./leak-detection");
Object.defineProperty(exports, "setCurrentRequestId", { enumerable: true, get: function () { return leak_detection_1.setCurrentRequestId; } });
Object.defineProperty(exports, "stopLeakDetection", { enumerable: true, get: function () { return leak_detection_1.stopLeakDetection; } });
__exportStar(require("./provider-implementation"), exports);
const response_stream_1 = require("./streaming/response-stream");
Object.defineProperty(exports, "ResponseStream", { enumerable: true, get: function () { return response_stream_1.ResponseStream; } });
const stream_writer_1 = require("./streaming/stream-writer");
Object.defineProperty(exports, "streamWriter", { enumerable: true, get: function () { return stream_writer_1.streamWriter; } });
const validate_composition_1 = require("./validate-composition");
Object.defineProperty(exports, "validateComposition", { enumerable: true, get: function () { return validate_composition_1.validateComposition; } });
