// @aws-sdk/client-service-quotas/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "servicequotas",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class ServiceQuotasClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultServiceQuotasHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class ServiceQuotasServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, ServiceQuotasServiceException.prototype);
    }
}

class AccessDeniedException extends ServiceQuotasServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
class AWSServiceAccessNotEnabledException extends ServiceQuotasServiceException {
    name = "AWSServiceAccessNotEnabledException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AWSServiceAccessNotEnabledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AWSServiceAccessNotEnabledException.prototype);
        this.Message = opts.Message;
    }
}
class DependencyAccessDeniedException extends ServiceQuotasServiceException {
    name = "DependencyAccessDeniedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DependencyAccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DependencyAccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
class NoAvailableOrganizationException extends ServiceQuotasServiceException {
    name = "NoAvailableOrganizationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoAvailableOrganizationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoAvailableOrganizationException.prototype);
        this.Message = opts.Message;
    }
}
class OrganizationNotInAllFeaturesModeException extends ServiceQuotasServiceException {
    name = "OrganizationNotInAllFeaturesModeException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OrganizationNotInAllFeaturesModeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OrganizationNotInAllFeaturesModeException.prototype);
        this.Message = opts.Message;
    }
}
class ServiceException extends ServiceQuotasServiceException {
    name = "ServiceException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "ServiceException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceException.prototype);
        this.Message = opts.Message;
    }
}
class TemplatesNotAvailableInRegionException extends ServiceQuotasServiceException {
    name = "TemplatesNotAvailableInRegionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TemplatesNotAvailableInRegionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TemplatesNotAvailableInRegionException.prototype);
        this.Message = opts.Message;
    }
}
class TooManyRequestsException extends ServiceQuotasServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
    }
}
class IllegalArgumentException extends ServiceQuotasServiceException {
    name = "IllegalArgumentException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalArgumentException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalArgumentException.prototype);
        this.Message = opts.Message;
    }
}
class InvalidResourceStateException extends ServiceQuotasServiceException {
    name = "InvalidResourceStateException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidResourceStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidResourceStateException.prototype);
        this.Message = opts.Message;
    }
}
class NoSuchResourceException extends ServiceQuotasServiceException {
    name = "NoSuchResourceException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchResourceException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchResourceException.prototype);
        this.Message = opts.Message;
    }
}
class ResourceAlreadyExistsException extends ServiceQuotasServiceException {
    name = "ResourceAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
class ServiceQuotaTemplateNotInUseException extends ServiceQuotasServiceException {
    name = "ServiceQuotaTemplateNotInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ServiceQuotaTemplateNotInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaTemplateNotInUseException.prototype);
        this.Message = opts.Message;
    }
}
class InvalidPaginationTokenException extends ServiceQuotasServiceException {
    name = "InvalidPaginationTokenException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidPaginationTokenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidPaginationTokenException.prototype);
        this.Message = opts.Message;
    }
}
class QuotaExceededException extends ServiceQuotasServiceException {
    name = "QuotaExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "QuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QuotaExceededException.prototype);
        this.Message = opts.Message;
    }
}
class TagPolicyViolationException extends ServiceQuotasServiceException {
    name = "TagPolicyViolationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TagPolicyViolationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TagPolicyViolationException.prototype);
        this.Message = opts.Message;
    }
}
class TooManyTagsException extends ServiceQuotasServiceException {
    name = "TooManyTagsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.Message = opts.Message;
    }
}

const _A = "Adjustable";
const _ADE = "AccessDeniedException";
const _AR = "AwsRegion";
const _ASQT = "AssociateServiceQuotaTemplate";
const _ASQTR = "AssociateServiceQuotaTemplateRequest";
const _ASQTRs = "AssociateServiceQuotaTemplateResponse";
const _AV = "AppliedValue";
const _AWSSANEE = "AWSServiceAccessNotEnabledException";
const _C = "Created";
const _CI = "ContextId";
const _CIa = "CaseId";
const _CS = "ContextScope";
const _CSC = "CreateSupportCase";
const _CSCR = "CreateSupportCaseRequest";
const _CSCRr = "CreateSupportCaseResponse";
const _CST = "ContextScopeType";
const _D = "Description";
const _DADE = "DependencyAccessDeniedException";
const _DSQIRFT = "DeleteServiceQuotaIncreaseRequestFromTemplate";
const _DSQIRFTR = "DeleteServiceQuotaIncreaseRequestFromTemplateRequest";
const _DSQIRFTRe = "DeleteServiceQuotaIncreaseRequestFromTemplateResponse";
const _DSQT = "DisassociateServiceQuotaTemplate";
const _DSQTR = "DisassociateServiceQuotaTemplateRequest";
const _DSQTRi = "DisassociateServiceQuotaTemplateResponse";
const _DV = "DesiredValue";
const _DVe = "DefaultValue";
const _EC = "ErrorCode";
const _EL = "ExclusionList";
const _EM = "ErrorMessage";
const _EQL = "ExclusionQuotaList";
const _ER = "ErrorReason";
const _GA = "GeneratedAt";
const _GAFSQT = "GetAssociationForServiceQuotaTemplate";
const _GAFSQTR = "GetAssociationForServiceQuotaTemplateRequest";
const _GAFSQTRe = "GetAssociationForServiceQuotaTemplateResponse";
const _GAMC = "GetAutoManagementConfiguration";
const _GAMCR = "GetAutoManagementConfigurationRequest";
const _GAMCRe = "GetAutoManagementConfigurationResponse";
const _GAWSDSQ = "GetAWSDefaultServiceQuota";
const _GAWSDSQR = "GetAWSDefaultServiceQuotaRequest";
const _GAWSDSQRe = "GetAWSDefaultServiceQuotaResponse";
const _GQ = "GlobalQuota";
const _GQUR = "GetQuotaUtilizationReport";
const _GQURR = "GetQuotaUtilizationReportRequest";
const _GQURRe = "GetQuotaUtilizationReportResponse";
const _GRSQC = "GetRequestedServiceQuotaChange";
const _GRSQCR = "GetRequestedServiceQuotaChangeRequest";
const _GRSQCRe = "GetRequestedServiceQuotaChangeResponse";
const _GSQ = "GetServiceQuota";
const _GSQIRFT = "GetServiceQuotaIncreaseRequestFromTemplate";
const _GSQIRFTR = "GetServiceQuotaIncreaseRequestFromTemplateRequest";
const _GSQIRFTRe = "GetServiceQuotaIncreaseRequestFromTemplateResponse";
const _GSQR = "GetServiceQuotaRequest";
const _GSQRe = "GetServiceQuotaResponse";
const _I = "Id";
const _IAE = "IllegalArgumentException";
const _IPTE = "InvalidPaginationTokenException";
const _IRSE = "InvalidResourceStateException";
const _IT = "InputTags";
const _K = "Key";
const _LAWSDSQ = "ListAWSDefaultServiceQuotas";
const _LAWSDSQR = "ListAWSDefaultServiceQuotasRequest";
const _LAWSDSQRi = "ListAWSDefaultServiceQuotasResponse";
const _LRSQCH = "ListRequestedServiceQuotaChangeHistory";
const _LRSQCHBQ = "ListRequestedServiceQuotaChangeHistoryByQuota";
const _LRSQCHBQR = "ListRequestedServiceQuotaChangeHistoryByQuotaRequest";
const _LRSQCHBQRi = "ListRequestedServiceQuotaChangeHistoryByQuotaResponse";
const _LRSQCHR = "ListRequestedServiceQuotaChangeHistoryRequest";
const _LRSQCHRi = "ListRequestedServiceQuotaChangeHistoryResponse";
const _LS = "ListServices";
const _LSQ = "ListServiceQuotas";
const _LSQIRIT = "ListServiceQuotaIncreaseRequestsInTemplate";
const _LSQIRITR = "ListServiceQuotaIncreaseRequestsInTemplateRequest";
const _LSQIRITRi = "ListServiceQuotaIncreaseRequestsInTemplateResponse";
const _LSQR = "ListServiceQuotasRequest";
const _LSQRi = "ListServiceQuotasResponse";
const _LSR = "ListServicesRequest";
const _LSRi = "ListServicesResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LU = "LastUpdated";
const _M = "Message";
const _MD = "MetricDimensions";
const _MI = "MetricInfo";
const _MN = "MetricNamespace";
const _MNe = "MetricName";
const _MR = "MaxResults";
const _MSR = "MetricStatisticRecommendation";
const _N = "Namespace";
const _NA = "NotificationArn";
const _NAOE = "NoAvailableOrganizationException";
const _NSRE = "NoSuchResourceException";
const _NT = "NextToken";
const _OIL = "OptInLevel";
const _OIS = "OptInStatus";
const _OIT = "OptInType";
const _ONIAFME = "OrganizationNotInAllFeaturesModeException";
const _OT = "OutputTags";
const _P = "Period";
const _PSQIRIT = "PutServiceQuotaIncreaseRequestIntoTemplate";
const _PSQIRITR = "PutServiceQuotaIncreaseRequestIntoTemplateRequest";
const _PSQIRITRu = "PutServiceQuotaIncreaseRequestIntoTemplateResponse";
const _PU = "PeriodUnit";
const _PV = "PeriodValue";
const _Q = "Quota";
const _QA = "QuotaArn";
const _QAAL = "QuotaAppliedAtLevel";
const _QC = "QuotaCode";
const _QCI = "QuotaContextInfo";
const _QCu = "QuotaContext";
const _QEE = "QuotaExceededException";
const _QI = "QuotaInfo";
const _QIL = "QuotaInfoList";
const _QN = "QuotaName";
const _QP = "QuotaPeriod";
const _QRAL = "QuotaRequestedAtLevel";
const _QUI = "QuotaUtilizationInfo";
const _QUIL = "QuotaUtilizationInfoList";
const _Qu = "Quotas";
const _R = "Requester";
const _RAEE = "ResourceAlreadyExistsException";
const _RARN = "ResourceARN";
const _RI = "RequestId";
const _RIe = "ReportId";
const _RQ = "RequestedQuota";
const _RQe = "RequestedQuotas";
const _RSQC = "RequestedServiceQuotaChange";
const _RSQCHLD = "RequestedServiceQuotaChangeHistoryListDefinition";
const _RSQI = "RequestServiceQuotaIncrease";
const _RSQIR = "RequestServiceQuotaIncreaseRequest";
const _RSQIRe = "RequestServiceQuotaIncreaseResponse";
const _RT = "RequestType";
const _S = "Status";
const _SAM = "StartAutoManagement";
const _SAMR = "StartAutoManagementRequest";
const _SAMRt = "StartAutoManagementResponse";
const _SAMRto = "StopAutoManagementRequest";
const _SAMRtop = "StopAutoManagementResponse";
const _SAMt = "StopAutoManagement";
const _SC = "ServiceCode";
const _SCA = "SupportCaseAllowed";
const _SE = "ServiceException";
const _SI = "ServiceInfo";
const _SILD = "ServiceInfoListDefinition";
const _SN = "ServiceName";
const _SQ = "ServiceQuota";
const _SQIRIT = "ServiceQuotaIncreaseRequestInTemplate";
const _SQIRITL = "ServiceQuotaIncreaseRequestInTemplateList";
const _SQLD = "ServiceQuotaListDefinition";
const _SQTAS = "ServiceQuotaTemplateAssociationStatus";
const _SQTNIUE = "ServiceQuotaTemplateNotInUseException";
const _SQUR = "StartQuotaUtilizationReport";
const _SQURR = "StartQuotaUtilizationReportRequest";
const _SQURRt = "StartQuotaUtilizationReportResponse";
const _Se = "Services";
const _T = "Tags";
const _TC = "TotalCount";
const _TK = "TagKeys";
const _TMRE = "TooManyRequestsException";
const _TMTE = "TooManyTagsException";
const _TNAIRE = "TemplatesNotAvailableInRegionException";
const _TPVE = "TagPolicyViolationException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tag";
const _U = "Utilization";
const _UAM = "UpdateAutoManagement";
const _UAMR = "UpdateAutoManagementRequest";
const _UAMRp = "UpdateAutoManagementResponse";
const _UM = "UsageMetric";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _Un = "Unit";
const _V = "Value";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _s = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.servicequotas";
const n0 = "com.amazonaws.servicequotas";
var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(AccessDeniedException$, AccessDeniedException);
var AssociateServiceQuotaTemplateRequest$ = [3, n0, _ASQTR,
    0,
    [],
    []
];
var AssociateServiceQuotaTemplateResponse$ = [3, n0, _ASQTRs,
    0,
    [],
    []
];
var AWSServiceAccessNotEnabledException$ = [-3, n0, _AWSSANEE,
    { [_e]: _c, [_hE]: 403 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(AWSServiceAccessNotEnabledException$, AWSServiceAccessNotEnabledException);
var CreateSupportCaseRequest$ = [3, n0, _CSCR,
    0,
    [_RI],
    [0], 1
];
var CreateSupportCaseResponse$ = [3, n0, _CSCRr,
    0,
    [],
    []
];
var DeleteServiceQuotaIncreaseRequestFromTemplateRequest$ = [3, n0, _DSQIRFTR,
    0,
    [_SC, _QC, _AR],
    [0, 0, 0], 3
];
var DeleteServiceQuotaIncreaseRequestFromTemplateResponse$ = [3, n0, _DSQIRFTRe,
    0,
    [],
    []
];
var DependencyAccessDeniedException$ = [-3, n0, _DADE,
    { [_e]: _c, [_hE]: 403 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(DependencyAccessDeniedException$, DependencyAccessDeniedException);
var DisassociateServiceQuotaTemplateRequest$ = [3, n0, _DSQTR,
    0,
    [],
    []
];
var DisassociateServiceQuotaTemplateResponse$ = [3, n0, _DSQTRi,
    0,
    [],
    []
];
var ErrorReason$ = [3, n0, _ER,
    0,
    [_EC, _EM],
    [0, 0]
];
var GetAssociationForServiceQuotaTemplateRequest$ = [3, n0, _GAFSQTR,
    0,
    [],
    []
];
var GetAssociationForServiceQuotaTemplateResponse$ = [3, n0, _GAFSQTRe,
    0,
    [_SQTAS],
    [0]
];
var GetAutoManagementConfigurationRequest$ = [3, n0, _GAMCR,
    0,
    [],
    []
];
var GetAutoManagementConfigurationResponse$ = [3, n0, _GAMCRe,
    0,
    [_OIL, _OIT, _NA, _OIS, _EL],
    [0, 0, 0, 0, () => ExclusionQuotaList]
];
var GetAWSDefaultServiceQuotaRequest$ = [3, n0, _GAWSDSQR,
    0,
    [_SC, _QC],
    [0, 0], 2
];
var GetAWSDefaultServiceQuotaResponse$ = [3, n0, _GAWSDSQRe,
    0,
    [_Q],
    [() => ServiceQuota$]
];
var GetQuotaUtilizationReportRequest$ = [3, n0, _GQURR,
    0,
    [_RIe, _NT, _MR],
    [0, 0, 1], 1
];
var GetQuotaUtilizationReportResponse$ = [3, n0, _GQURRe,
    0,
    [_RIe, _S, _GA, _TC, _Qu, _NT, _EC, _EM],
    [0, 0, 4, 1, () => QuotaUtilizationInfoList, 0, 0, 0]
];
var GetRequestedServiceQuotaChangeRequest$ = [3, n0, _GRSQCR,
    0,
    [_RI],
    [0], 1
];
var GetRequestedServiceQuotaChangeResponse$ = [3, n0, _GRSQCRe,
    0,
    [_RQ],
    [() => RequestedServiceQuotaChange$]
];
var GetServiceQuotaIncreaseRequestFromTemplateRequest$ = [3, n0, _GSQIRFTR,
    0,
    [_SC, _QC, _AR],
    [0, 0, 0], 3
];
var GetServiceQuotaIncreaseRequestFromTemplateResponse$ = [3, n0, _GSQIRFTRe,
    0,
    [_SQIRIT],
    [() => ServiceQuotaIncreaseRequestInTemplate$]
];
var GetServiceQuotaRequest$ = [3, n0, _GSQR,
    0,
    [_SC, _QC, _CI],
    [0, 0, 0], 2
];
var GetServiceQuotaResponse$ = [3, n0, _GSQRe,
    0,
    [_Q],
    [() => ServiceQuota$]
];
var IllegalArgumentException$ = [-3, n0, _IAE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(IllegalArgumentException$, IllegalArgumentException);
var InvalidPaginationTokenException$ = [-3, n0, _IPTE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidPaginationTokenException$, InvalidPaginationTokenException);
var InvalidResourceStateException$ = [-3, n0, _IRSE,
    { [_e]: _c, [_hE]: 405 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidResourceStateException$, InvalidResourceStateException);
var ListAWSDefaultServiceQuotasRequest$ = [3, n0, _LAWSDSQR,
    0,
    [_SC, _NT, _MR],
    [0, 0, 1], 1
];
var ListAWSDefaultServiceQuotasResponse$ = [3, n0, _LAWSDSQRi,
    0,
    [_NT, _Qu],
    [0, () => ServiceQuotaListDefinition]
];
var ListRequestedServiceQuotaChangeHistoryByQuotaRequest$ = [3, n0, _LRSQCHBQR,
    0,
    [_SC, _QC, _S, _NT, _MR, _QRAL],
    [0, 0, 0, 0, 1, 0], 2
];
var ListRequestedServiceQuotaChangeHistoryByQuotaResponse$ = [3, n0, _LRSQCHBQRi,
    0,
    [_NT, _RQe],
    [0, () => RequestedServiceQuotaChangeHistoryListDefinition]
];
var ListRequestedServiceQuotaChangeHistoryRequest$ = [3, n0, _LRSQCHR,
    0,
    [_SC, _S, _NT, _MR, _QRAL],
    [0, 0, 0, 1, 0]
];
var ListRequestedServiceQuotaChangeHistoryResponse$ = [3, n0, _LRSQCHRi,
    0,
    [_NT, _RQe],
    [0, () => RequestedServiceQuotaChangeHistoryListDefinition]
];
var ListServiceQuotaIncreaseRequestsInTemplateRequest$ = [3, n0, _LSQIRITR,
    0,
    [_SC, _AR, _NT, _MR],
    [0, 0, 0, 1]
];
var ListServiceQuotaIncreaseRequestsInTemplateResponse$ = [3, n0, _LSQIRITRi,
    0,
    [_SQIRITL, _NT],
    [() => ServiceQuotaIncreaseRequestInTemplateList, 0]
];
var ListServiceQuotasRequest$ = [3, n0, _LSQR,
    0,
    [_SC, _NT, _MR, _QC, _QAAL],
    [0, 0, 1, 0, 0], 1
];
var ListServiceQuotasResponse$ = [3, n0, _LSQRi,
    0,
    [_NT, _Qu],
    [0, () => ServiceQuotaListDefinition]
];
var ListServicesRequest$ = [3, n0, _LSR,
    0,
    [_NT, _MR],
    [0, 1]
];
var ListServicesResponse$ = [3, n0, _LSRi,
    0,
    [_NT, _Se],
    [0, () => ServiceInfoListDefinition]
];
var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_T],
    [() => OutputTags]
];
var MetricInfo$ = [3, n0, _MI,
    0,
    [_MN, _MNe, _MD, _MSR],
    [0, 0, 128 | 0, 0]
];
var NoAvailableOrganizationException$ = [-3, n0, _NAOE,
    { [_e]: _c, [_hE]: 403 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(NoAvailableOrganizationException$, NoAvailableOrganizationException);
var NoSuchResourceException$ = [-3, n0, _NSRE,
    { [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(NoSuchResourceException$, NoSuchResourceException);
var OrganizationNotInAllFeaturesModeException$ = [-3, n0, _ONIAFME,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(OrganizationNotInAllFeaturesModeException$, OrganizationNotInAllFeaturesModeException);
var PutServiceQuotaIncreaseRequestIntoTemplateRequest$ = [3, n0, _PSQIRITR,
    0,
    [_QC, _SC, _AR, _DV],
    [0, 0, 0, 1], 4
];
var PutServiceQuotaIncreaseRequestIntoTemplateResponse$ = [3, n0, _PSQIRITRu,
    0,
    [_SQIRIT],
    [() => ServiceQuotaIncreaseRequestInTemplate$]
];
var QuotaContextInfo$ = [3, n0, _QCI,
    0,
    [_CS, _CST, _CI],
    [0, 0, 0]
];
var QuotaExceededException$ = [-3, n0, _QEE,
    { [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(QuotaExceededException$, QuotaExceededException);
var QuotaInfo$ = [3, n0, _QI,
    0,
    [_QC, _QN],
    [0, 0]
];
var QuotaPeriod$ = [3, n0, _QP,
    0,
    [_PV, _PU],
    [1, 0]
];
var QuotaUtilizationInfo$ = [3, n0, _QUI,
    0,
    [_QC, _SC, _QN, _N, _U, _DVe, _AV, _SN, _A],
    [0, 0, 0, 0, 1, 1, 1, 0, 2]
];
var RequestedServiceQuotaChange$ = [3, n0, _RSQC,
    0,
    [_I, _RT, _CIa, _SC, _SN, _QC, _QN, _DV, _S, _C, _LU, _R, _QA, _GQ, _Un, _QRAL, _QCu],
    [0, 0, 0, 0, 0, 0, 0, 1, 0, 4, 4, 0, 0, 2, 0, 0, () => QuotaContextInfo$]
];
var RequestServiceQuotaIncreaseRequest$ = [3, n0, _RSQIR,
    0,
    [_SC, _QC, _DV, _CI, _SCA],
    [0, 0, 1, 0, 2], 3
];
var RequestServiceQuotaIncreaseResponse$ = [3, n0, _RSQIRe,
    0,
    [_RQ],
    [() => RequestedServiceQuotaChange$]
];
var ResourceAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ResourceAlreadyExistsException$, ResourceAlreadyExistsException);
var ServiceException$ = [-3, n0, _SE,
    { [_e]: _s, [_hE]: 500 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceException$, ServiceException);
var ServiceInfo$ = [3, n0, _SI,
    0,
    [_SC, _SN],
    [0, 0]
];
var ServiceQuota$ = [3, n0, _SQ,
    0,
    [_SC, _SN, _QA, _QC, _QN, _V, _Un, _A, _GQ, _UM, _P, _ER, _QAAL, _QCu, _D],
    [0, 0, 0, 0, 0, 1, 0, 2, 2, () => MetricInfo$, () => QuotaPeriod$, () => ErrorReason$, 0, () => QuotaContextInfo$, 0]
];
var ServiceQuotaIncreaseRequestInTemplate$ = [3, n0, _SQIRIT,
    0,
    [_SC, _SN, _QC, _QN, _DV, _AR, _Un, _GQ],
    [0, 0, 0, 0, 1, 0, 0, 2]
];
var ServiceQuotaTemplateNotInUseException$ = [-3, n0, _SQTNIUE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceQuotaTemplateNotInUseException$, ServiceQuotaTemplateNotInUseException);
var StartAutoManagementRequest$ = [3, n0, _SAMR,
    0,
    [_OIL, _OIT, _NA, _EL],
    [0, 0, 0, [2, n0, _EL, 0, 0, 64 | 0]], 2
];
var StartAutoManagementResponse$ = [3, n0, _SAMRt,
    0,
    [],
    []
];
var StartQuotaUtilizationReportRequest$ = [3, n0, _SQURR,
    0,
    [],
    []
];
var StartQuotaUtilizationReportResponse$ = [3, n0, _SQURRt,
    0,
    [_RIe, _S, _M],
    [0, 0, 0]
];
var StopAutoManagementRequest$ = [3, n0, _SAMRto,
    0,
    [],
    []
];
var StopAutoManagementResponse$ = [3, n0, _SAMRtop,
    0,
    [],
    []
];
var Tag$ = [3, n0, _Ta,
    0,
    [_K, _V],
    [0, 0], 2
];
var TagPolicyViolationException$ = [-3, n0, _TPVE,
    { [_e]: _c, [_hE]: 401 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(TagPolicyViolationException$, TagPolicyViolationException);
var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _T],
    [0, () => InputTags], 2
];
var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
var TemplatesNotAvailableInRegionException$ = [-3, n0, _TNAIRE,
    { [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(TemplatesNotAvailableInRegionException$, TemplatesNotAvailableInRegionException);
var TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c, [_hE]: 429 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(TooManyRequestsException$, TooManyRequestsException);
var TooManyTagsException$ = [-3, n0, _TMTE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(TooManyTagsException$, TooManyTagsException);
var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
var UpdateAutoManagementRequest$ = [3, n0, _UAMR,
    0,
    [_OIT, _NA, _EL],
    [0, 0, [2, n0, _EL, 0, 0, 64 | 0]]
];
var UpdateAutoManagementResponse$ = [3, n0, _UAMRp,
    0,
    [],
    []
];
var ServiceQuotasServiceException$ = [-3, _sm, "ServiceQuotasServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(ServiceQuotasServiceException$, ServiceQuotasServiceException);
var InputTags = [1, n0, _IT,
    0, () => Tag$
];
var OutputTags = [1, n0, _OT,
    0, () => Tag$
];
var QuotaInfoList = [1, n0, _QIL,
    0, () => QuotaInfo$
];
var QuotaUtilizationInfoList = [1, n0, _QUIL,
    0, () => QuotaUtilizationInfo$
];
var RequestedServiceQuotaChangeHistoryListDefinition = [1, n0, _RSQCHLD,
    0, () => RequestedServiceQuotaChange$
];
var ServiceInfoListDefinition = [1, n0, _SILD,
    0, () => ServiceInfo$
];
var ServiceQuotaIncreaseRequestInTemplateList = [1, n0, _SQIRITL,
    0, () => ServiceQuotaIncreaseRequestInTemplate$
];
var ServiceQuotaListDefinition = [1, n0, _SQLD,
    0, () => ServiceQuota$
];
var ExclusionQuotaList = [2, n0, _EQL,
    0, 0, () => QuotaInfoList
];
var AssociateServiceQuotaTemplate$ = [9, n0, _ASQT,
    0, () => AssociateServiceQuotaTemplateRequest$, () => AssociateServiceQuotaTemplateResponse$
];
var CreateSupportCase$ = [9, n0, _CSC,
    0, () => CreateSupportCaseRequest$, () => CreateSupportCaseResponse$
];
var DeleteServiceQuotaIncreaseRequestFromTemplate$ = [9, n0, _DSQIRFT,
    0, () => DeleteServiceQuotaIncreaseRequestFromTemplateRequest$, () => DeleteServiceQuotaIncreaseRequestFromTemplateResponse$
];
var DisassociateServiceQuotaTemplate$ = [9, n0, _DSQT,
    0, () => DisassociateServiceQuotaTemplateRequest$, () => DisassociateServiceQuotaTemplateResponse$
];
var GetAssociationForServiceQuotaTemplate$ = [9, n0, _GAFSQT,
    0, () => GetAssociationForServiceQuotaTemplateRequest$, () => GetAssociationForServiceQuotaTemplateResponse$
];
var GetAutoManagementConfiguration$ = [9, n0, _GAMC,
    0, () => GetAutoManagementConfigurationRequest$, () => GetAutoManagementConfigurationResponse$
];
var GetAWSDefaultServiceQuota$ = [9, n0, _GAWSDSQ,
    0, () => GetAWSDefaultServiceQuotaRequest$, () => GetAWSDefaultServiceQuotaResponse$
];
var GetQuotaUtilizationReport$ = [9, n0, _GQUR,
    0, () => GetQuotaUtilizationReportRequest$, () => GetQuotaUtilizationReportResponse$
];
var GetRequestedServiceQuotaChange$ = [9, n0, _GRSQC,
    0, () => GetRequestedServiceQuotaChangeRequest$, () => GetRequestedServiceQuotaChangeResponse$
];
var GetServiceQuota$ = [9, n0, _GSQ,
    0, () => GetServiceQuotaRequest$, () => GetServiceQuotaResponse$
];
var GetServiceQuotaIncreaseRequestFromTemplate$ = [9, n0, _GSQIRFT,
    0, () => GetServiceQuotaIncreaseRequestFromTemplateRequest$, () => GetServiceQuotaIncreaseRequestFromTemplateResponse$
];
var ListAWSDefaultServiceQuotas$ = [9, n0, _LAWSDSQ,
    0, () => ListAWSDefaultServiceQuotasRequest$, () => ListAWSDefaultServiceQuotasResponse$
];
var ListRequestedServiceQuotaChangeHistory$ = [9, n0, _LRSQCH,
    0, () => ListRequestedServiceQuotaChangeHistoryRequest$, () => ListRequestedServiceQuotaChangeHistoryResponse$
];
var ListRequestedServiceQuotaChangeHistoryByQuota$ = [9, n0, _LRSQCHBQ,
    0, () => ListRequestedServiceQuotaChangeHistoryByQuotaRequest$, () => ListRequestedServiceQuotaChangeHistoryByQuotaResponse$
];
var ListServiceQuotaIncreaseRequestsInTemplate$ = [9, n0, _LSQIRIT,
    0, () => ListServiceQuotaIncreaseRequestsInTemplateRequest$, () => ListServiceQuotaIncreaseRequestsInTemplateResponse$
];
var ListServiceQuotas$ = [9, n0, _LSQ,
    0, () => ListServiceQuotasRequest$, () => ListServiceQuotasResponse$
];
var ListServices$ = [9, n0, _LS,
    0, () => ListServicesRequest$, () => ListServicesResponse$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
var PutServiceQuotaIncreaseRequestIntoTemplate$ = [9, n0, _PSQIRIT,
    0, () => PutServiceQuotaIncreaseRequestIntoTemplateRequest$, () => PutServiceQuotaIncreaseRequestIntoTemplateResponse$
];
var RequestServiceQuotaIncrease$ = [9, n0, _RSQI,
    0, () => RequestServiceQuotaIncreaseRequest$, () => RequestServiceQuotaIncreaseResponse$
];
var StartAutoManagement$ = [9, n0, _SAM,
    0, () => StartAutoManagementRequest$, () => StartAutoManagementResponse$
];
var StartQuotaUtilizationReport$ = [9, n0, _SQUR,
    0, () => StartQuotaUtilizationReportRequest$, () => StartQuotaUtilizationReportResponse$
];
var StopAutoManagement$ = [9, n0, _SAMt,
    0, () => StopAutoManagementRequest$, () => StopAutoManagementResponse$
];
var TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
var UpdateAutoManagement$ = [9, n0, _UAM,
    0, () => UpdateAutoManagementRequest$, () => UpdateAutoManagementResponse$
];

class AssociateServiceQuotaTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "AssociateServiceQuotaTemplate", {})
    .n("ServiceQuotasClient", "AssociateServiceQuotaTemplateCommand")
    .sc(AssociateServiceQuotaTemplate$)
    .build() {
}

class CreateSupportCaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "CreateSupportCase", {})
    .n("ServiceQuotasClient", "CreateSupportCaseCommand")
    .sc(CreateSupportCase$)
    .build() {
}

class DeleteServiceQuotaIncreaseRequestFromTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "DeleteServiceQuotaIncreaseRequestFromTemplate", {})
    .n("ServiceQuotasClient", "DeleteServiceQuotaIncreaseRequestFromTemplateCommand")
    .sc(DeleteServiceQuotaIncreaseRequestFromTemplate$)
    .build() {
}

class DisassociateServiceQuotaTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "DisassociateServiceQuotaTemplate", {})
    .n("ServiceQuotasClient", "DisassociateServiceQuotaTemplateCommand")
    .sc(DisassociateServiceQuotaTemplate$)
    .build() {
}

class GetAssociationForServiceQuotaTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetAssociationForServiceQuotaTemplate", {})
    .n("ServiceQuotasClient", "GetAssociationForServiceQuotaTemplateCommand")
    .sc(GetAssociationForServiceQuotaTemplate$)
    .build() {
}

class GetAutoManagementConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetAutoManagementConfiguration", {})
    .n("ServiceQuotasClient", "GetAutoManagementConfigurationCommand")
    .sc(GetAutoManagementConfiguration$)
    .build() {
}

class GetAWSDefaultServiceQuotaCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetAWSDefaultServiceQuota", {})
    .n("ServiceQuotasClient", "GetAWSDefaultServiceQuotaCommand")
    .sc(GetAWSDefaultServiceQuota$)
    .build() {
}

class GetQuotaUtilizationReportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetQuotaUtilizationReport", {})
    .n("ServiceQuotasClient", "GetQuotaUtilizationReportCommand")
    .sc(GetQuotaUtilizationReport$)
    .build() {
}

class GetRequestedServiceQuotaChangeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetRequestedServiceQuotaChange", {})
    .n("ServiceQuotasClient", "GetRequestedServiceQuotaChangeCommand")
    .sc(GetRequestedServiceQuotaChange$)
    .build() {
}

class GetServiceQuotaCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetServiceQuota", {})
    .n("ServiceQuotasClient", "GetServiceQuotaCommand")
    .sc(GetServiceQuota$)
    .build() {
}

class GetServiceQuotaIncreaseRequestFromTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "GetServiceQuotaIncreaseRequestFromTemplate", {})
    .n("ServiceQuotasClient", "GetServiceQuotaIncreaseRequestFromTemplateCommand")
    .sc(GetServiceQuotaIncreaseRequestFromTemplate$)
    .build() {
}

class ListAWSDefaultServiceQuotasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListAWSDefaultServiceQuotas", {})
    .n("ServiceQuotasClient", "ListAWSDefaultServiceQuotasCommand")
    .sc(ListAWSDefaultServiceQuotas$)
    .build() {
}

class ListRequestedServiceQuotaChangeHistoryByQuotaCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListRequestedServiceQuotaChangeHistoryByQuota", {})
    .n("ServiceQuotasClient", "ListRequestedServiceQuotaChangeHistoryByQuotaCommand")
    .sc(ListRequestedServiceQuotaChangeHistoryByQuota$)
    .build() {
}

class ListRequestedServiceQuotaChangeHistoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListRequestedServiceQuotaChangeHistory", {})
    .n("ServiceQuotasClient", "ListRequestedServiceQuotaChangeHistoryCommand")
    .sc(ListRequestedServiceQuotaChangeHistory$)
    .build() {
}

class ListServiceQuotaIncreaseRequestsInTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListServiceQuotaIncreaseRequestsInTemplate", {})
    .n("ServiceQuotasClient", "ListServiceQuotaIncreaseRequestsInTemplateCommand")
    .sc(ListServiceQuotaIncreaseRequestsInTemplate$)
    .build() {
}

class ListServiceQuotasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListServiceQuotas", {})
    .n("ServiceQuotasClient", "ListServiceQuotasCommand")
    .sc(ListServiceQuotas$)
    .build() {
}

class ListServicesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListServices", {})
    .n("ServiceQuotasClient", "ListServicesCommand")
    .sc(ListServices$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "ListTagsForResource", {})
    .n("ServiceQuotasClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class PutServiceQuotaIncreaseRequestIntoTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "PutServiceQuotaIncreaseRequestIntoTemplate", {})
    .n("ServiceQuotasClient", "PutServiceQuotaIncreaseRequestIntoTemplateCommand")
    .sc(PutServiceQuotaIncreaseRequestIntoTemplate$)
    .build() {
}

class RequestServiceQuotaIncreaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "RequestServiceQuotaIncrease", {})
    .n("ServiceQuotasClient", "RequestServiceQuotaIncreaseCommand")
    .sc(RequestServiceQuotaIncrease$)
    .build() {
}

class StartAutoManagementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "StartAutoManagement", {})
    .n("ServiceQuotasClient", "StartAutoManagementCommand")
    .sc(StartAutoManagement$)
    .build() {
}

class StartQuotaUtilizationReportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "StartQuotaUtilizationReport", {})
    .n("ServiceQuotasClient", "StartQuotaUtilizationReportCommand")
    .sc(StartQuotaUtilizationReport$)
    .build() {
}

class StopAutoManagementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "StopAutoManagement", {})
    .n("ServiceQuotasClient", "StopAutoManagementCommand")
    .sc(StopAutoManagement$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "TagResource", {})
    .n("ServiceQuotasClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "UntagResource", {})
    .n("ServiceQuotasClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateAutoManagementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ServiceQuotasV20190624", "UpdateAutoManagement", {})
    .n("ServiceQuotasClient", "UpdateAutoManagementCommand")
    .sc(UpdateAutoManagement$)
    .build() {
}

const paginateListAWSDefaultServiceQuotas = core.createPaginator(ServiceQuotasClient, ListAWSDefaultServiceQuotasCommand, "NextToken", "NextToken", "MaxResults");

const paginateListRequestedServiceQuotaChangeHistoryByQuota = core.createPaginator(ServiceQuotasClient, ListRequestedServiceQuotaChangeHistoryByQuotaCommand, "NextToken", "NextToken", "MaxResults");

const paginateListRequestedServiceQuotaChangeHistory = core.createPaginator(ServiceQuotasClient, ListRequestedServiceQuotaChangeHistoryCommand, "NextToken", "NextToken", "MaxResults");

const paginateListServiceQuotaIncreaseRequestsInTemplate = core.createPaginator(ServiceQuotasClient, ListServiceQuotaIncreaseRequestsInTemplateCommand, "NextToken", "NextToken", "MaxResults");

const paginateListServiceQuotas = core.createPaginator(ServiceQuotasClient, ListServiceQuotasCommand, "NextToken", "NextToken", "MaxResults");

const paginateListServices = core.createPaginator(ServiceQuotasClient, ListServicesCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AssociateServiceQuotaTemplateCommand,
    CreateSupportCaseCommand,
    DeleteServiceQuotaIncreaseRequestFromTemplateCommand,
    DisassociateServiceQuotaTemplateCommand,
    GetAssociationForServiceQuotaTemplateCommand,
    GetAutoManagementConfigurationCommand,
    GetAWSDefaultServiceQuotaCommand,
    GetQuotaUtilizationReportCommand,
    GetRequestedServiceQuotaChangeCommand,
    GetServiceQuotaCommand,
    GetServiceQuotaIncreaseRequestFromTemplateCommand,
    ListAWSDefaultServiceQuotasCommand,
    ListRequestedServiceQuotaChangeHistoryCommand,
    ListRequestedServiceQuotaChangeHistoryByQuotaCommand,
    ListServiceQuotaIncreaseRequestsInTemplateCommand,
    ListServiceQuotasCommand,
    ListServicesCommand,
    ListTagsForResourceCommand,
    PutServiceQuotaIncreaseRequestIntoTemplateCommand,
    RequestServiceQuotaIncreaseCommand,
    StartAutoManagementCommand,
    StartQuotaUtilizationReportCommand,
    StopAutoManagementCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateAutoManagementCommand,
};
const paginators = {
    paginateListAWSDefaultServiceQuotas,
    paginateListRequestedServiceQuotaChangeHistory,
    paginateListRequestedServiceQuotaChangeHistoryByQuota,
    paginateListServiceQuotaIncreaseRequestsInTemplate,
    paginateListServiceQuotas,
    paginateListServices,
};
class ServiceQuotas extends ServiceQuotasClient {
}
smithyClient.createAggregatedClient(commands, ServiceQuotas, { paginators });

const AppliedLevelEnum = {
    ACCOUNT: "ACCOUNT",
    ALL: "ALL",
    RESOURCE: "RESOURCE",
};
const ErrorCode = {
    DEPENDENCY_ACCESS_DENIED_ERROR: "DEPENDENCY_ACCESS_DENIED_ERROR",
    DEPENDENCY_SERVICE_ERROR: "DEPENDENCY_SERVICE_ERROR",
    DEPENDENCY_THROTTLING_ERROR: "DEPENDENCY_THROTTLING_ERROR",
    SERVICE_QUOTA_NOT_AVAILABLE_ERROR: "SERVICE_QUOTA_NOT_AVAILABLE_ERROR",
};
const ServiceQuotaTemplateAssociationStatus = {
    ASSOCIATED: "ASSOCIATED",
    DISASSOCIATED: "DISASSOCIATED",
};
const OptInLevel = {
    ACCOUNT: "ACCOUNT",
};
const OptInStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const OptInType = {
    NotifyAndAdjust: "NotifyAndAdjust",
    NotifyOnly: "NotifyOnly",
};
const PeriodUnit = {
    DAY: "DAY",
    HOUR: "HOUR",
    MICROSECOND: "MICROSECOND",
    MILLISECOND: "MILLISECOND",
    MINUTE: "MINUTE",
    SECOND: "SECOND",
    WEEK: "WEEK",
};
const QuotaContextScope = {
    ACCOUNT: "ACCOUNT",
    RESOURCE: "RESOURCE",
};
const ReportStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
};
const RequestType = {
    AutomaticManagement: "AutomaticManagement",
};
const RequestStatus = {
    APPROVED: "APPROVED",
    CASE_CLOSED: "CASE_CLOSED",
    CASE_OPENED: "CASE_OPENED",
    DENIED: "DENIED",
    INVALID_REQUEST: "INVALID_REQUEST",
    NOT_APPROVED: "NOT_APPROVED",
    PENDING: "PENDING",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AWSServiceAccessNotEnabledException = AWSServiceAccessNotEnabledException;
exports.AWSServiceAccessNotEnabledException$ = AWSServiceAccessNotEnabledException$;
exports.AccessDeniedException = AccessDeniedException;
exports.AccessDeniedException$ = AccessDeniedException$;
exports.AppliedLevelEnum = AppliedLevelEnum;
exports.AssociateServiceQuotaTemplate$ = AssociateServiceQuotaTemplate$;
exports.AssociateServiceQuotaTemplateCommand = AssociateServiceQuotaTemplateCommand;
exports.AssociateServiceQuotaTemplateRequest$ = AssociateServiceQuotaTemplateRequest$;
exports.AssociateServiceQuotaTemplateResponse$ = AssociateServiceQuotaTemplateResponse$;
exports.CreateSupportCase$ = CreateSupportCase$;
exports.CreateSupportCaseCommand = CreateSupportCaseCommand;
exports.CreateSupportCaseRequest$ = CreateSupportCaseRequest$;
exports.CreateSupportCaseResponse$ = CreateSupportCaseResponse$;
exports.DeleteServiceQuotaIncreaseRequestFromTemplate$ = DeleteServiceQuotaIncreaseRequestFromTemplate$;
exports.DeleteServiceQuotaIncreaseRequestFromTemplateCommand = DeleteServiceQuotaIncreaseRequestFromTemplateCommand;
exports.DeleteServiceQuotaIncreaseRequestFromTemplateRequest$ = DeleteServiceQuotaIncreaseRequestFromTemplateRequest$;
exports.DeleteServiceQuotaIncreaseRequestFromTemplateResponse$ = DeleteServiceQuotaIncreaseRequestFromTemplateResponse$;
exports.DependencyAccessDeniedException = DependencyAccessDeniedException;
exports.DependencyAccessDeniedException$ = DependencyAccessDeniedException$;
exports.DisassociateServiceQuotaTemplate$ = DisassociateServiceQuotaTemplate$;
exports.DisassociateServiceQuotaTemplateCommand = DisassociateServiceQuotaTemplateCommand;
exports.DisassociateServiceQuotaTemplateRequest$ = DisassociateServiceQuotaTemplateRequest$;
exports.DisassociateServiceQuotaTemplateResponse$ = DisassociateServiceQuotaTemplateResponse$;
exports.ErrorCode = ErrorCode;
exports.ErrorReason$ = ErrorReason$;
exports.GetAWSDefaultServiceQuota$ = GetAWSDefaultServiceQuota$;
exports.GetAWSDefaultServiceQuotaCommand = GetAWSDefaultServiceQuotaCommand;
exports.GetAWSDefaultServiceQuotaRequest$ = GetAWSDefaultServiceQuotaRequest$;
exports.GetAWSDefaultServiceQuotaResponse$ = GetAWSDefaultServiceQuotaResponse$;
exports.GetAssociationForServiceQuotaTemplate$ = GetAssociationForServiceQuotaTemplate$;
exports.GetAssociationForServiceQuotaTemplateCommand = GetAssociationForServiceQuotaTemplateCommand;
exports.GetAssociationForServiceQuotaTemplateRequest$ = GetAssociationForServiceQuotaTemplateRequest$;
exports.GetAssociationForServiceQuotaTemplateResponse$ = GetAssociationForServiceQuotaTemplateResponse$;
exports.GetAutoManagementConfiguration$ = GetAutoManagementConfiguration$;
exports.GetAutoManagementConfigurationCommand = GetAutoManagementConfigurationCommand;
exports.GetAutoManagementConfigurationRequest$ = GetAutoManagementConfigurationRequest$;
exports.GetAutoManagementConfigurationResponse$ = GetAutoManagementConfigurationResponse$;
exports.GetQuotaUtilizationReport$ = GetQuotaUtilizationReport$;
exports.GetQuotaUtilizationReportCommand = GetQuotaUtilizationReportCommand;
exports.GetQuotaUtilizationReportRequest$ = GetQuotaUtilizationReportRequest$;
exports.GetQuotaUtilizationReportResponse$ = GetQuotaUtilizationReportResponse$;
exports.GetRequestedServiceQuotaChange$ = GetRequestedServiceQuotaChange$;
exports.GetRequestedServiceQuotaChangeCommand = GetRequestedServiceQuotaChangeCommand;
exports.GetRequestedServiceQuotaChangeRequest$ = GetRequestedServiceQuotaChangeRequest$;
exports.GetRequestedServiceQuotaChangeResponse$ = GetRequestedServiceQuotaChangeResponse$;
exports.GetServiceQuota$ = GetServiceQuota$;
exports.GetServiceQuotaCommand = GetServiceQuotaCommand;
exports.GetServiceQuotaIncreaseRequestFromTemplate$ = GetServiceQuotaIncreaseRequestFromTemplate$;
exports.GetServiceQuotaIncreaseRequestFromTemplateCommand = GetServiceQuotaIncreaseRequestFromTemplateCommand;
exports.GetServiceQuotaIncreaseRequestFromTemplateRequest$ = GetServiceQuotaIncreaseRequestFromTemplateRequest$;
exports.GetServiceQuotaIncreaseRequestFromTemplateResponse$ = GetServiceQuotaIncreaseRequestFromTemplateResponse$;
exports.GetServiceQuotaRequest$ = GetServiceQuotaRequest$;
exports.GetServiceQuotaResponse$ = GetServiceQuotaResponse$;
exports.IllegalArgumentException = IllegalArgumentException;
exports.IllegalArgumentException$ = IllegalArgumentException$;
exports.InvalidPaginationTokenException = InvalidPaginationTokenException;
exports.InvalidPaginationTokenException$ = InvalidPaginationTokenException$;
exports.InvalidResourceStateException = InvalidResourceStateException;
exports.InvalidResourceStateException$ = InvalidResourceStateException$;
exports.ListAWSDefaultServiceQuotas$ = ListAWSDefaultServiceQuotas$;
exports.ListAWSDefaultServiceQuotasCommand = ListAWSDefaultServiceQuotasCommand;
exports.ListAWSDefaultServiceQuotasRequest$ = ListAWSDefaultServiceQuotasRequest$;
exports.ListAWSDefaultServiceQuotasResponse$ = ListAWSDefaultServiceQuotasResponse$;
exports.ListRequestedServiceQuotaChangeHistory$ = ListRequestedServiceQuotaChangeHistory$;
exports.ListRequestedServiceQuotaChangeHistoryByQuota$ = ListRequestedServiceQuotaChangeHistoryByQuota$;
exports.ListRequestedServiceQuotaChangeHistoryByQuotaCommand = ListRequestedServiceQuotaChangeHistoryByQuotaCommand;
exports.ListRequestedServiceQuotaChangeHistoryByQuotaRequest$ = ListRequestedServiceQuotaChangeHistoryByQuotaRequest$;
exports.ListRequestedServiceQuotaChangeHistoryByQuotaResponse$ = ListRequestedServiceQuotaChangeHistoryByQuotaResponse$;
exports.ListRequestedServiceQuotaChangeHistoryCommand = ListRequestedServiceQuotaChangeHistoryCommand;
exports.ListRequestedServiceQuotaChangeHistoryRequest$ = ListRequestedServiceQuotaChangeHistoryRequest$;
exports.ListRequestedServiceQuotaChangeHistoryResponse$ = ListRequestedServiceQuotaChangeHistoryResponse$;
exports.ListServiceQuotaIncreaseRequestsInTemplate$ = ListServiceQuotaIncreaseRequestsInTemplate$;
exports.ListServiceQuotaIncreaseRequestsInTemplateCommand = ListServiceQuotaIncreaseRequestsInTemplateCommand;
exports.ListServiceQuotaIncreaseRequestsInTemplateRequest$ = ListServiceQuotaIncreaseRequestsInTemplateRequest$;
exports.ListServiceQuotaIncreaseRequestsInTemplateResponse$ = ListServiceQuotaIncreaseRequestsInTemplateResponse$;
exports.ListServiceQuotas$ = ListServiceQuotas$;
exports.ListServiceQuotasCommand = ListServiceQuotasCommand;
exports.ListServiceQuotasRequest$ = ListServiceQuotasRequest$;
exports.ListServiceQuotasResponse$ = ListServiceQuotasResponse$;
exports.ListServices$ = ListServices$;
exports.ListServicesCommand = ListServicesCommand;
exports.ListServicesRequest$ = ListServicesRequest$;
exports.ListServicesResponse$ = ListServicesResponse$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
exports.MetricInfo$ = MetricInfo$;
exports.NoAvailableOrganizationException = NoAvailableOrganizationException;
exports.NoAvailableOrganizationException$ = NoAvailableOrganizationException$;
exports.NoSuchResourceException = NoSuchResourceException;
exports.NoSuchResourceException$ = NoSuchResourceException$;
exports.OptInLevel = OptInLevel;
exports.OptInStatus = OptInStatus;
exports.OptInType = OptInType;
exports.OrganizationNotInAllFeaturesModeException = OrganizationNotInAllFeaturesModeException;
exports.OrganizationNotInAllFeaturesModeException$ = OrganizationNotInAllFeaturesModeException$;
exports.PeriodUnit = PeriodUnit;
exports.PutServiceQuotaIncreaseRequestIntoTemplate$ = PutServiceQuotaIncreaseRequestIntoTemplate$;
exports.PutServiceQuotaIncreaseRequestIntoTemplateCommand = PutServiceQuotaIncreaseRequestIntoTemplateCommand;
exports.PutServiceQuotaIncreaseRequestIntoTemplateRequest$ = PutServiceQuotaIncreaseRequestIntoTemplateRequest$;
exports.PutServiceQuotaIncreaseRequestIntoTemplateResponse$ = PutServiceQuotaIncreaseRequestIntoTemplateResponse$;
exports.QuotaContextInfo$ = QuotaContextInfo$;
exports.QuotaContextScope = QuotaContextScope;
exports.QuotaExceededException = QuotaExceededException;
exports.QuotaExceededException$ = QuotaExceededException$;
exports.QuotaInfo$ = QuotaInfo$;
exports.QuotaPeriod$ = QuotaPeriod$;
exports.QuotaUtilizationInfo$ = QuotaUtilizationInfo$;
exports.ReportStatus = ReportStatus;
exports.RequestServiceQuotaIncrease$ = RequestServiceQuotaIncrease$;
exports.RequestServiceQuotaIncreaseCommand = RequestServiceQuotaIncreaseCommand;
exports.RequestServiceQuotaIncreaseRequest$ = RequestServiceQuotaIncreaseRequest$;
exports.RequestServiceQuotaIncreaseResponse$ = RequestServiceQuotaIncreaseResponse$;
exports.RequestStatus = RequestStatus;
exports.RequestType = RequestType;
exports.RequestedServiceQuotaChange$ = RequestedServiceQuotaChange$;
exports.ResourceAlreadyExistsException = ResourceAlreadyExistsException;
exports.ResourceAlreadyExistsException$ = ResourceAlreadyExistsException$;
exports.ServiceException = ServiceException;
exports.ServiceException$ = ServiceException$;
exports.ServiceInfo$ = ServiceInfo$;
exports.ServiceQuota$ = ServiceQuota$;
exports.ServiceQuotaIncreaseRequestInTemplate$ = ServiceQuotaIncreaseRequestInTemplate$;
exports.ServiceQuotaTemplateAssociationStatus = ServiceQuotaTemplateAssociationStatus;
exports.ServiceQuotaTemplateNotInUseException = ServiceQuotaTemplateNotInUseException;
exports.ServiceQuotaTemplateNotInUseException$ = ServiceQuotaTemplateNotInUseException$;
exports.ServiceQuotas = ServiceQuotas;
exports.ServiceQuotasClient = ServiceQuotasClient;
exports.ServiceQuotasServiceException = ServiceQuotasServiceException;
exports.ServiceQuotasServiceException$ = ServiceQuotasServiceException$;
exports.StartAutoManagement$ = StartAutoManagement$;
exports.StartAutoManagementCommand = StartAutoManagementCommand;
exports.StartAutoManagementRequest$ = StartAutoManagementRequest$;
exports.StartAutoManagementResponse$ = StartAutoManagementResponse$;
exports.StartQuotaUtilizationReport$ = StartQuotaUtilizationReport$;
exports.StartQuotaUtilizationReportCommand = StartQuotaUtilizationReportCommand;
exports.StartQuotaUtilizationReportRequest$ = StartQuotaUtilizationReportRequest$;
exports.StartQuotaUtilizationReportResponse$ = StartQuotaUtilizationReportResponse$;
exports.StopAutoManagement$ = StopAutoManagement$;
exports.StopAutoManagementCommand = StopAutoManagementCommand;
exports.StopAutoManagementRequest$ = StopAutoManagementRequest$;
exports.StopAutoManagementResponse$ = StopAutoManagementResponse$;
exports.Tag$ = Tag$;
exports.TagPolicyViolationException = TagPolicyViolationException;
exports.TagPolicyViolationException$ = TagPolicyViolationException$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceRequest$ = TagResourceRequest$;
exports.TagResourceResponse$ = TagResourceResponse$;
exports.TemplatesNotAvailableInRegionException = TemplatesNotAvailableInRegionException;
exports.TemplatesNotAvailableInRegionException$ = TemplatesNotAvailableInRegionException$;
exports.TooManyRequestsException = TooManyRequestsException;
exports.TooManyRequestsException$ = TooManyRequestsException$;
exports.TooManyTagsException = TooManyTagsException;
exports.TooManyTagsException$ = TooManyTagsException$;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceRequest$ = UntagResourceRequest$;
exports.UntagResourceResponse$ = UntagResourceResponse$;
exports.UpdateAutoManagement$ = UpdateAutoManagement$;
exports.UpdateAutoManagementCommand = UpdateAutoManagementCommand;
exports.UpdateAutoManagementRequest$ = UpdateAutoManagementRequest$;
exports.UpdateAutoManagementResponse$ = UpdateAutoManagementResponse$;
exports.paginateListAWSDefaultServiceQuotas = paginateListAWSDefaultServiceQuotas;
exports.paginateListRequestedServiceQuotaChangeHistory = paginateListRequestedServiceQuotaChangeHistory;
exports.paginateListRequestedServiceQuotaChangeHistoryByQuota = paginateListRequestedServiceQuotaChangeHistoryByQuota;
exports.paginateListServiceQuotaIncreaseRequestsInTemplate = paginateListServiceQuotaIncreaseRequestsInTemplate;
exports.paginateListServiceQuotas = paginateListServiceQuotas;
exports.paginateListServices = paginateListServices;
